qiskit/circuit/quantumcircuit.py

```diff
@@ -1997,72 +1997,78 @@ def depth(
         filter_function: Optional[callable] = lambda x: not getattr(
             x.operation, "_directive", False
         ),
+        recurse: bool = True,
     ) -> int:
         """Return circuit depth (i.e., length of critical path).
 
         Args:
             filter_function (callable): a function to filter out some instructions.
                 Should take as input a tuple of (Instruction, list(Qubit), list(Clbit)).
                 By default filters out "directives", such as barrier or snapshot.
+            recurse: if ``True`` (default), then recurse into control-flow operations.  For loops
+                with known-length iterators are counted unrolled.  If-else blocks take the
+                longer case of the two branches.  While loops are counted as if the condition runs
+                once only.
 
         Returns:
             int: Depth of circuit.
 
         Notes:
-            The circuit depth and the DAG depth need not be the
-            same.
+            The circuit depth and the DAG depth need not be the same.
         """
-        # Assign each bit in the circuit a unique integer
-        # to index into op_stack.
-        bit_indices = {bit: idx for idx, bit in enumerate(self.qubits + self.clbits)}
-
-        # If no bits, return 0
-        if not bit_indices:
-            return 0
-
-        # A list that holds the height of each qubit
-        # and classical bit.
-        op_stack = [0] * len(bit_indices)
-
-        # Here we are playing a modified version of
-        # Tetris where we stack gates, but multi-qubit
-        # gates, or measurements have a block for each
-        # qubit or cbit that are connected by a virtual
-        # line so that they all stacked at the same depth.
-        # Conditional gates act on all cbits in the register
-        # they are conditioned on.
-        # The max stack height is the circuit depth.
-        for instruction in self._data:
-            levels = []
-            reg_ints = []
-            for ind, reg in enumerate(instruction.qubits + instruction.clbits):
-                # Add to the stacks of the qubits and
-                # cbits used in the gate.
-                reg_ints.append(bit_indices[reg])
-                if filter_function(instruction):
-                    levels.append(op_stack[reg_ints[ind]] + 1)
+        # pylint: disable=cyclic-import
+        from qiskit.circuit.controlflow import ControlFlowOp, ForLoopOp
+
+        def _depth_inner(circuit, weight=1, wire_map=None):
```

Comment:
Could you add a comment about these parameters here?

Reply:
Done in d52669f4c.